src/runtime_src/core/pcie/linux/system_linux.cpp

```diff
@@ -168,12 +147,24 @@ get_num_dev_total(bool is_user) const
 system_linux::
 system_linux()
 {
-  if (singleton)
-    throw std::runtime_error("More than one SHIM registered!");
-  singleton = this;
+  // Add built-in driver to the list.
+  pci::register_driver(std::make_shared<pci::drv_xocl>());
+  pci::register_driver(std::make_shared<pci::drv_xclmgmt>());
```

Comment:
It would make the code more readable if this just directly emplaced the driver_list.  I was scratching my head figuring out how the driver_list (iterated below) was populated.

Reply:
Sure. Also changed the comments a little bit. Hope it will make code more clearer.